internal/repository/limits/events.go

```diff
@@ -52,10 +53,25 @@ func ChangeAuditLogRetention(auditLogRetention *time.Duration) LimitsChange {
 	}
 }
 
+func ChangeDisallowPublicOrgRegistration(disallow *bool) LimitsChange {
+	return func(e *SetEvent) {
+		e.DisallowPublicOrgRegistration = disallow
+	}
+}
+
 var SetEventMapper = eventstore.GenericEventMapper[SetEvent]
 
+type ResetProperty int
+
+const (
+	ResetUnknownProperty ResetProperty = iota
+	ResetAuditLogRetention
+	ResetAllowPublicOrgRegistration
+)
+
 type ResetEvent struct {
 	*eventstore.BaseEvent `json:"-"`
+	Properties            []ResetProperty `json:"properties,omitempty"`
```

Comment:
if the reset event depending on the contents of this slice either deletes all limits or changes only something, then I would split the ResetEvent into a change and a reset/remove event

Reply:
Would you be fine if we renamed it to ResetOnly?

Reply:
Would be an option, IMO the intention of this events have to be a little clearer, as it deletes all limits if you provide no properties at all at the moment.

Reply:
obsolete, as we have no reset event anymore